apps/web/src/dialogs/command-palette/command-palette-dialog.tsx

```diff
@@ -529,3 +531,27 @@ function getCommandPaletteHelp(isCommandMode: boolean) {
         ])
   ];
 }
+
+/**
+ * commands need to be sorted wrt groups,
+ * meaning commands of same group should be next to each other,
+ * and recent commands should be at the top
+ */
+function sortCommands(commands: Command[]) {
+  const recent: Command[] = [];
+  const sortedWrtGroups: Command[][] = [];
+  for (const command of commands) {
+    const group = command.group;
+    if (group === "recent") {
+      recent.push(command);
+      continue;
+    }
+    const index = sortedWrtGroups.findIndex((c) => c[0].group === group);
+    if (index === -1) {
+      sortedWrtGroups.push([command]);
+    } else {
+      sortedWrtGroups[index].push(command);
+    }
+  }
+  return recent.concat(sortedWrtGroups.flat());
+}
```

Comment:
You can just `.sort` by `group` instead of this custom logic.

Reply:
I'm facing a challenge where `.sort` respects the order of the searched elements, need to figure this out

Reply:
Were you able to figure something out? If not, we can keep this I think.

Reply:
Nope